flake.nix

```diff
@@ -0,0 +1,27 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = {nixpkgs, ...}: let
+    system = "x86_64-linux";
+    pkgs = nixpkgs.legacyPackages.${system};
+  in {
+    devShells.${system}.default = pkgs.mkShell {
+      nativeBuildInputs = with pkgs; [
+        cargo-binstall
+        dioxus-cli
+        wasm-bindgen-cli_0_2_100
+      ];
+
+      # https://docs.rs/getrandom/0.3.3/getrandom/#webassembly-support
+      RUSTFLAGS = "--cfg getrandom_backend=\"wasm_js\"";
+
+      # gcc is default
```

Comment:
suggestion (blocking): I did some more research but couldn't really figure out why gcc doesn't work and where I got the information from that clang does work. but maybe we can still be a bit more detailed here:

```suggestion
      # with gcc (default), the WASM build fails silently
      # at runtime there is an error message similar to https://github.com/DioxusLabs/dioxus/discussions/3807
      # maybe this can be resolved with dioxus 0.7.0 due to https://github.com/DioxusLabs/dioxus/issues/4023
```